File: server/src/main/java/com/preponderous/roam/model/entity/Chicken.java
package com.preponderous.roam.model.entity;

import com.preponderous.roam.model.LivingEntity;

/**
 * Represents a chicken wildlife entity.
 * Chickens are passive animals that flee from threats.
 * 
 * @author Daniel McCoy Stephenson
 */
public class Chicken extends LivingEntity {
    public static final double DEFAULT_ENERGY = 30.0;
    public static final int DEFAULT_MOVE_SPEED = 25;
    public static final double FLEE_RANGE = 3.0;
    
    private int moveSpeed;
    private double fleeRange;

    public Chicken(long tickCreated) {
        super("Chicken", "assets/images/chicken.png", DEFAULT_ENERGY, tickCreated);
        this.moveSpeed = DEFAULT_MOVE_SPEED;
        this.fleeRange = FLEE_RANGE;
        this.setSolid(false);
    }
    
    /**
     * Constructor for persistence - creates chicken with specific ID.
     */
    public Chicken(String id, long tickCreated) {
        super(id, "Chicken", "assets/images/chicken.png", DEFAULT_ENERGY, tickCreated);
        this.moveSpeed = DEFAULT_MOVE_SPEED;
        this.fleeRange = FLEE_RANGE;
        this.setSolid(false);
    }

    public int getMoveSpeed() {
        return moveSpeed;
    }

    public void setMoveSpeed(int moveSpeed) {
        this.moveSpeed = moveSpeed;
    }

    public double getFleeRange() {
        return fleeRange;
    }

    public void setFleeRange(double fleeRange) {
        this.fleeRange = fleeRange;
    }
}

File: src/entity/living/chicken.py
import random
from entity.grass import Grass
from entity.living.livingEntity import LivingEntity


# @author Daniel McCoy Stephenson
# @since July 7th, 2022
class Chicken(LivingEntity):
    def __init__(self, tickCreated):
        LivingEntity.__init__(
            self,
            "Chicken",
            "assets/images/chicken.png",
            random.randrange(20, 30),
            [Grass],
            tickCreated,
        )

    def getDefaultImagePath(self):
        return "assets/images/chicken.png"

    def getReproductionCooldownImagePath(self):
        return "assets/images/chickenOnReproductionCooldown.png"

    def createOffspring(self, tick):
        return Chicken(tick)
